frontend/src/components/UploadModal/UploadModal.js

```diff
@@ -0,0 +1,197 @@
+import { formatBytesToString } from '../../util';
+
+export default function UploadModal({ show, onUpload, onClose }) {
+  const [addedVideos, setAddedVideos] = useState([]);
+  const [videoNames, setVideoNames] = useState([]);
+  const [showSnackbar, setShowSnackbar] = useState({
+    noVideosAdded: false,
+    videoNameMissing: false,
+  });
+
+  const handleClose = () => {
+    setAddedVideos([]);
+    setVideoNames([]);
+    onClose();
+  };
+
+  const handleUpload = () => {
+    if (addedVideos.length === 0) {
+      setShowSnackbar({ ...showSnackbar, noVideosAdded: true });
+      return;
+    }
+    if (videoNames.some(name => name === '')) {
+      setShowSnackbar({ ...showSnackbar, videoNameMissing: true });
+      return;
+    }
+    onUpload(addedVideos, videoNames);
+    handleClose();
+  }
+
+  const handleAddVideo = (videos) => {
+    setShowSnackbar({ ...showSnackbar, noVideosAdded: false });
+    setAddedVideos([...addedVideos, ...videos]);
+    setVideoNames([...videoNames, ...videos.map(video => /[^.]*/.exec(video.file.name)[0])]);
+  }
+
+  const getVideoAddedMessage = (videoName) => `Video ${videoName} was successfully added. `;
+
+  const handleChangeVideoName = (videoIndex) => (
+    (event) => {
+      const videoNamesCopy = videoNames.slice();
+      videoNamesCopy.splice(videoIndex, 1, event.target.value);
+      setVideoNames([...videoNamesCopy]);
+      setShowSnackbar({ ...showSnackbar, videoNameMissing: false });
+    }
+  );
+
+  const handleRemoveVideo = (videoIndex) => (
+    () => {
+      const videoNamesCopy = videoNames.slice();
+      videoNamesCopy.splice(videoIndex, 1);
+      const addedVideosCopy = addedVideos.slice();
+      addedVideosCopy.splice(videoIndex, 1);
+      setVideoNames([...videoNamesCopy]);
+      setAddedVideos([...addedVideosCopy]);
+    }
+  )
+
+  const renderAddedVideosList = () => {
+    const listItems = addedVideos.map((video, index) => {
+      const videoName = videoNames[index];
+      return (
+        <ListItem key={video.file.name + index.toString()}>
+          <TextField
+            fullWidth
+            error={videoName === ''}
+            helperText={
+              videoName === '' ? 'Please enter a name for this video' : ''
+            }
+            value={videoName}
+            onChange={handleChangeVideoName(index)}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  {formatBytesToString(video.file.size)}
+                  <IconButton onClick={handleRemoveVideo(index)}>
+                    <RemoveIcon />
+                  </IconButton>
+                </InputAdornment>
+              ),
+            }}
+          />
+        </ListItem>
+      );
+    });
+
+    return (
+      <List>
+        {listItems}
+      </List>
+    );
+  }
+
+  const renderSnackbarsIfNeeded = () => {
+    if (showSnackbar.noVideosAdded) {
+      return (
+        <CustomSnackbar
```

Comment:
General comment about snackbars. I tried adding Lord of The rings: Fellowship of the Ring BluRay Extended eddition, which is rather large and there was no notification about the process. I think there should be a snackbar that stays on which says something like "video(s) is being added" because the user is just kept hanging without any information. Also, while this is true, upload button should be disabled